vendor/github.com/samuel/go-zookeeper/zk/dnshostprovider.go

```diff
@@ -0,0 +1,88 @@
+package zk
+
+import (
+	"fmt"
+	"net"
+	"sync"
+)
+
+// DNSHostProvider is the default HostProvider. It currently matches
+// the Java StaticHostProvider, resolving hosts from DNS once during
+// the call to Init.  It could be easily extended to re-query DNS
+// periodically or if there is trouble connecting.
+type DNSHostProvider struct {
```

Comment:
I suppose nothing stops us from writing our own HostProvider implementation
that panic's or fatal's if there's problems. Is that what you're aiming for?

On Fri, Sep 9, 2016 at 8:11 AM, Sargun Dhillon <notifications@github.com>
wrote:

> In vendor/github.com/samuel/go-zookeeper/zk/dnshostprovider.go
> <https://github.com/mesosphere/mesos-dns/pull/470#discussion_r78169189>:
>
> > @@ -0,0 +1,88 @@
> > +package zk
> > +
> > +import (
> > +	"fmt"
> > +	"net"
> > +	"sync"
> > +)
> > +
> > +// DNSHostProvider is the default HostProvider. It currently matches
> > +// the Java StaticHostProvider, resolving hosts from DNS once during
> > +// the call to Init.  It could be easily extended to re-query DNS
> > +// periodically or if there is trouble connecting.
> > +type DNSHostProvider struct {
>
> I wish we could have the behaviour that if ZK fails, we recreate the
> client.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/mesosphere/mesos-dns/pull/470/files/76cb2deb1618862fa8f3595590f6f22cffe958a1#r78169189>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ACPVLNN0JCtk7RfRtvAeJWcyvyAEFO8Uks5qoUzwgaJpZM4J2Q4l>
> .
>